autoload/clang_format.vim

```diff
@@ -238,6 +239,13 @@ function! clang_format#replace(line1, line2, ...) abort
 
     let winview = winsaveview()
     let splitted = split(formatted, '\n', 1)
+    if getline(a:line1, a:line2) ==# splitted[a:line1-1:a:line2-1]
+        if g:clang_format#praise
+            echo "No formatting needed, looking fabulous already!"
+        endif
+        " Early out, no need to introduce a change
+        return
+    endif
 
```

Comment:
Should probably clear here too. Imagine the case of no format and then formatting.